Dispatcher/UI/Screens/HomePage/HomePageViewController.swift

```diff
@@ -33,14 +40,19 @@ extension HomePageViewController: UITableViewDataSource, UITableViewDelegate {
     func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
         let cell = tableView.dequeueReusableCell(withIdentifier: ArticleCell.identifier, for: indexPath)
         as? ArticleCell
-        let item = viewModel.articles[indexPath.row]
-        cell?.initCell(article: item as? Article)
+        let newArticle = viewModel.articles[indexPath.row] as? NewsArticle
+        cell?.initCell(with: newArticle)
+        setImage(cell,url:URL(string: newArticle?.urlToImage ?? ""))
```

Comment:
too much is going on in this line for just one line in function

it would be better to do this inside the cell

cell.setImage()

the cell already knows the url with its data